packages/client/components/promptResponse/EmojiMenuTipTap.tsx

```diff
@@ -0,0 +1,73 @@
+import {Editor, Range} from '@tiptap/core'
+import Suggestion from '@tiptap/suggestion'
+import {PluginKey} from 'prosemirror-state'
+import React, {useEffect, useState} from 'react'
+import EmojiMenuContainer from '../TaskEditor/EmojiMenuContainer'
+import {getSelectionBoundingBox} from './tiptapConfig'
+
+interface Props {
+  tiptapEditor: Editor
+}
+
+const pluginKey = new PluginKey('emojiMenu')
+
+const EmojiMenuTipTap = (props: Props) => {
+  const {tiptapEditor} = props
+  const [openEmojiMenu, setOpenEmojiMenu] = useState(false)
+  const [emojiQuery, setEmojiQuery] = useState('')
+  const [range, setRange] = useState<Range | null>(null)
+
+  useEffect(() => {
+    if (tiptapEditor.isDestroyed) {
+      return
+    }
+
+    const plugin = Suggestion({
+      pluginKey: pluginKey as any,
```

Comment:
+1 just curious, why do we need to cast here?

Reply:
The types between the `PluginKey` I'm using and the `PluginKey` `Suggestion` expects don't quite match up exactly, so we need to cast to work around that.

Added a comment + TODO for this.